### ISSUE TITLE
Types issue: grouping by added field

### ISSUE THREAD

**Issue by soanvig** [koskimas/kysely#132]
If I have table let's say `{ tableName: { column1: string }}` then in `.groupBy` I can only group by `column1` string, which is fine.

However if I add something to a query via `.select().as('column2')` then I cannot group by this column.

This would be effectively something like this SQL:

```sql
SELECT 'Text' as field, COUNT(*) GROUP BY field;
```

In postgres this works, but I cannot to make it working in Kysely types. I have to cast as any (or just `sql.ref`):

```ts
.groupBy([sql.ref('field')])
```

Looks like a bug to me or maybe it's postgres specific that's why it is not implemented.

**Comment by koskimas** [comment]
Yep, that's a bug. I'll fix it right away. Thanks for opening the issue!

**Comment by koskimas** [comment]
Fixed in 0.21.1

**Comment by soanvig** [comment]
Fell in love in this library, very well designed in my opinion :)
